linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java

```diff
@@ -1158,4 +1161,72 @@ private static void deleteAllFiles(FileSystem fileSystem, FsPath fsPath) throws
     }
     fileSystem.delete(fsPath);
   }
+
+  @ApiOperation(value = "chmod", notes = "file chmod", response = Message.class)
+  @ApiImplicitParams({
+    @ApiImplicitParam(name = "filepath", required = false, dataType = "String", value = "filepath"),
+    @ApiImplicitParam(
+        name = "isRecursion",
+        required = false,
+        dataType = "String",
+        value = "isRecursion"),
+    @ApiImplicitParam(
+        name = "filePermission",
+        required = false,
+        dataType = "String",
+        value = "filePermission"),
+  })
+  @RequestMapping(path = "/chmod", method = RequestMethod.GET)
+  public Message chmod(
+      HttpServletRequest req,
+      @RequestParam(value = "filepath", required = true) String filePath,
+      @RequestParam(value = "isRecursion", required = false, defaultValue = "true") Boolean isRecursion,
+      @RequestParam(value = "filePermission", required = true) String filePermission)
+      throws WorkSpaceException, IOException {
+    String userName = ModuleUserUtils.getOperationUser(req, "chmod " + filePath);
+    if (StringUtils.isEmpty(filePath)) {
+      return Message.error(MessageFormat.format(PARAMETER_NOT_BLANK, filePath));
+    }
+    if (StringUtils.isEmpty(filePermission)) {
+      return Message.error(MessageFormat.format(PARAMETER_NOT_BLANK, filePermission));
+    } else {
+      Matcher matcher = Pattern.compile("[0-7]{3}").matcher(filePermission);
+      if (matcher.matches()) {
+        return Message.error(MessageFormat.format(PERMISSION_FORMAT_ERROR, filePermission));
+      }
+    }
```

Comment:
There's a slight problem with the logic. Please fix it.

Reply:
done